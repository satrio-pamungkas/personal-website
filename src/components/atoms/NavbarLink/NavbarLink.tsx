import Link from "next/link"

interface INavbarLink {
  label: string
  active: boolean
  link: string
}

export const NavbarLink = (props: INavbarLink) => {
  return (
    <Link 
      data-testid="link" 
      href={props.link} 
      className={`block py-2 pl-3 pr-4 text-white rounded md:bg-transparent md:hover:underline ` +
       `md:hover:underline-offset-8 md:text-white md:p-0 font-barlow uppercase hover:bg-neutral-900 ` +
       `${props.active ? 'font-bold' : 'font-regular'}`}
    >
      {props.label}
    </Link>
  );
}