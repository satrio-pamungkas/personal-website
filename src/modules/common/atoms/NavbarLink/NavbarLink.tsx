interface INavbarLink {
  label: string
}

export const NavbarLink = (props: INavbarLink) => {
  return (
    <a href="" className="block py-2 pl-3 pr-4 text-white 
      bg-blue-700 rounded md:bg-transparent md:text-white 
      md:p-0 font-barlow uppercase"
    >
      {props.label}
    </a>
  );
}