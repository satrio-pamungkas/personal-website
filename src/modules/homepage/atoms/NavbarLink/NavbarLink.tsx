interface INavbarLink {
  label: string
}

export const NavbarLink = (props: INavbarLink) => {
  return (
    <a>{props.label}</a>
  );
}