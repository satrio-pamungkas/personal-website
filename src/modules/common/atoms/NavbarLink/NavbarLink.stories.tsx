import { ComponentStory } from '@storybook/react';
import { NavbarLink } from "./NavbarLink";

export default {
  title: 'Common/Atoms/NavbarLink',
  component: NavbarLink
};

const Template: ComponentStory<typeof NavbarLink> = (args) => {
  return (<NavbarLink {...args} />)
}

export const Primary = Template.bind({});
Primary.args = { label: "Homepage", active: true }

export const Secondary = Template.bind({});
Secondary.args = { label: "Homepage", active: false }