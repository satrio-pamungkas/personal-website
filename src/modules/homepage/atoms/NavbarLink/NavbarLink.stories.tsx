import { ComponentStory } from '@storybook/react';
import { NavbarLink } from "./NavbarLink";

export default {
  title: 'NavbarLink',
  component: NavbarLink
};

export const Primary: ComponentStory<typeof NavbarLink> = () => { 
  return (<NavbarLink label="Homepage"/>)
}