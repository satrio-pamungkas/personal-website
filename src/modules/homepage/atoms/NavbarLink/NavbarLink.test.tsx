import renderer, { act } from 'react-test-renderer';
import { NavbarLink } from './NavbarLink';
import ReactDOM from 'react-dom/client';

describe("NavbarLink", () => { 
  it("Renders correctly", () => {
    const component = renderer.create(<NavbarLink label="Homepage"/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  }) 

  it("Renders with label name correctly", () => {
    const label = "Homepage";
    const component = (
      <NavbarLink label={label}/>
    );
    const container = document.createElement("a");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.createRoot(container).render(component);
    })
    expect(document.body.textContent).toContain(label);
  })
});