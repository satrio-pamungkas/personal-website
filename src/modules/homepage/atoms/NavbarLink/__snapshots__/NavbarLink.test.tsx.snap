// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink Renders correctly 1`] = `
<a>
  Homepage
</a>
`;
